### ISSUE TITLE
Regarding Code

### ISSUE THREAD

**Issue by Nenkai** [GrowtopiaNoobs/GrowtopiaServer#4]
https://github.com/GrowtopiaNoobs/GrowtopiaServer/blob/153bb967c7d00a06dcf865cf56670c67f4529df6/enet%20server%20test/enet%20server%20test.cpp#L1227

Obvious decompiling here. I suggest this to be changed as any copyrighted client code can actually be used in court. The rest of the code also hints to such conclusions, however as mentioned the code is messy so I haven't fully checked.

Offtopic: Grats on first open-sourced however i have to admit isn't the first as i have been working for one for almost a year, with all key components that are mentioned on your readme nearly done. Not planning to open-source it, but thought it'd be interesting to share.

**Comment by GrowtopiaNoobs** [comment]
Yes, there are some parts which have sign of reverse engineer. It is because I was writing this server fast and I couldn't focus on code style, but I think that this problem can't be fully fixed because packets still need to have growtopia style and contain some of growtopia text.

Btw do you know what is proper ways of destroying tiles and how to pick up items?

Also good job about your server :+1: You said that you have done most of stuff that I have said in my readme and most of that stuff is about performance, so I am interested how do you test performance and what do you use your server for? And how can you validate your threads are working well if you don't test your server under load?

Offtopic: And I guess that you made items.dat decoder, so I am interested if you will make that ever public or no? (I made my own yesterday. It still doesn't deocode items type, but decoding strings works well. I am not planning to release for now.)

**Comment by ritzysw** [comment]
1.deny that u are data miner as 'TK69' name and get all new buy+ worlds
2.privately make pserver by reverse engineering gt as 'Nenkai' name
Nice

**Comment by Nenkai** [comment]
> Yes, there are some parts which have sign of reverse engineer. It is because I was writing this server fast and I couldn't focus on code style, but I think that this problem can't be fully fixed because packets still need to have growtopia style and contain some of growtopia text.
> 
> Btw do you know what is proper ways of destroying tiles and how to pick up items?
> 
> Also good job about your server  You said that you have done most of stuff that I have said in my readme and most of that stuff is about performance, so I am interested how do you test performance and what do you use your server for? And how can you validate your threads are working well if you don't test your server under load?
> 
> Offtopic: And I guess that you made items.dat decoder, so I am interested if you will make that ever public or no? (I made my own yesterday. It still doesn't deocode items type, but decoding strings works well. I am not planning to release for now.)

Performance wasn't a real key components so most of the things done is threading networking for each peer. As for destroying tiles and picking up items, look into raw packets.
No I haven't made an items.dat decoder. Using the data being put on the wiki instead by another admin and converting it properly.

**Comment by ipr0gr4mipas2** [comment]
> > Yes, there are some parts which have sign of reverse engineer. It is because I was writing this server fast and I couldn't focus on code style, but I think that this problem can't be fully fixed because packets still need to have growtopia style and contain some of growtopia text.
> > Btw do you know what is proper ways of destroying tiles and how to pick up items?
> > Also good job about your server  You said that you have done most of stuff that I have said in my readme and most of that stuff is about performance, so I am interested how do you test performance and what do you use your server for? And how can you validate your threads are working well if you don't test your server under load?
> > Offtopic: And I guess that you made items.dat decoder, so I am interested if you will make that ever public or no? (I made my own yesterday. It still doesn't deocode items type, but decoding strings works well. I am not planning to release for now.)
> 
> Performance wasn't a real key components so most of the things done is threading networking for each peer. As for destroying tiles and picking up items, look into raw packets. For destroying tiles, it should emit a world tile change type packet. Picking items up should emit 3 of them - One for world change, one for item added, and one for backpack change.
> No I haven't made an items.dat decoder. Using the data being put on the wiki instead by another admin and converting it properly. That means, alot of fields are missing so currently using the same way as yours.
> 
> To reply to the offtopic question, no plans to do anything with it until main server shuts down.

Actually, it does not!
It emits a single packet (of PlayerMoving class type) that gets the pixel X and Y, as well as the UID in the client, and another #.

You could divide the positions by 32 to check the tile. However, you can't actually remove the thing dropped from my knowledge.

**Comment by Nenkai** [comment]
> > > Yes, there are some parts which have sign of reverse engineer. It is because I was writing this server fast and I couldn't focus on code style, but I think that this problem can't be fully fixed because packets still need to have growtopia style and contain some of growtopia text.
> > > Btw do you know what is proper ways of destroying tiles and how to pick up items?
> > > Also good job about your server  You said that you have done most of stuff that I have said in my readme and most of that stuff is about performance, so I am interested how do you test performance and what do you use your server for? And how can you validate your threads are working well if you don't test your server under load?
> > > Offtopic: And I guess that you made items.dat decoder, so I am interested if you will make that ever public or no? (I made my own yesterday. It still doesn't deocode items type, but decoding strings works well. I am not planning to release for now.)
> > 
> > 
> > Performance wasn't a real key components so most of the things done is threading networking for each peer. As for destroying tiles and picking up items, look into raw packets. For destroying tiles, it should emit a world tile change type packet. Picking items up should emit 3 of them - One for world change, one for item added, and one for backpack change.
> > No I haven't made an items.dat decoder. Using the data being put on the wiki instead by another admin and converting it properly. That means, alot of fields are missing so currently using the same way as yours.
> > To reply to the offtopic question, no plans to do anything with it until main server shuts down.
> 
> Actually, it does not!
> It emits a single packet (of PlayerMoving class type) that gets the pixel X and Y, as well as the UID in the client, and another #.
> 
> You could divide the positions by 32 to check the tile. However, you can't actually remove the thing dropped from my knowledge.

Thats just for punching. I'm sure its about destroying the tile itself.

**Comment by ritzysw** [comment]
currently it just sets broken tile to 0 and doesn't really break it

**Comment by GrowtopiaNoobs** [comment]
> > > > Yes, there are some parts which have sign of reverse engineer. It is because I was writing this server fast and I couldn't focus on code style, but I think that this problem can't be fully fixed because packets still need to have growtopia style and contain some of growtopia text.
> > > > Btw do you know what is proper ways of destroying tiles and how to pick up items?
> > > > Also good job about your server  You said that you have done most of stuff that I have said in my readme and most of that stuff is about performance, so I am interested how do you test performance and what do you use your server for? And how can you validate your threads are working well if you don't test your server under load?
> > > > Offtopic: And I guess that you made items.dat decoder, so I am interested if you will make that ever public or no? (I made my own yesterday. It still doesn't deocode items type, but decoding strings works well. I am not planning to release for now.)
> > > 
> > > 
> > > Performance wasn't a real key components so most of the things done is threading networking for each peer. As for destroying tiles and picking up items, look into raw packets. For destroying tiles, it should emit a world tile change type packet. Picking items up should emit 3 of them - One for world change, one for item added, and one for backpack change.
> > > No I haven't made an items.dat decoder. Using the data being put on the wiki instead by another admin and converting it properly. That means, alot of fields are missing so currently using the same way as yours.
> > > To reply to the offtopic question, no plans to do anything with it until main server shuts down.
> > 
> > 
> > Actually, it does not!
> > It emits a single packet (of PlayerMoving class type) that gets the pixel X and Y, as well as the UID in the client, and another #.
> > You could divide the positions by 32 to check the tile. However, you can't actually remove the thing dropped from my knowledge.
> 
> Thats just for punching. I'm sure its about destroying the tile itself.

I tried to look and for picking up item there is one packet (I am not talking about local player, cuz as Nenkai said, there else would be inventory change packet), but I haven't got it working yet. And I can confirm that when block is destructed then there is emmited one packet.

**Comment by ipr0gr4mipas2** [comment]
the dropped item thing is done on the server actually

so basically

client sends collect item packet
server looks into it and looks for hacks
is ok? send a collect item packet

**Comment by iytvu** [comment]
@GrowtopiaNoobs Join our Discord your welcomed legend ,
https://discord.gg/e3q87Gv

**Comment by iProgramMC** [comment]
Ah yes. The proper breaking sets 18 (Fist ID) instead of 0 or 6864 (which you use to break the blocks). As for the proper collecting you need to extend the world packet by at least 8 bytes (because that is where the dropped items are located)
